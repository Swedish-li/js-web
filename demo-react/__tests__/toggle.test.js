import React from 'react'
import renderer from 'react-test-renderer'
import Toggle from '../src/toggle'

test('app', () => {
  const component = renderer.create(
    <Toggle />
  )

  let tree = component.toJSON()
  //   { type: 'button',
  //   props: { onClick: [Function: bound handleClick] },
  //   children: [ 'ON' ] }
  console.log(tree)
  expect(tree).toMatchSnapshot()

  tree.props.onClick()
  tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
