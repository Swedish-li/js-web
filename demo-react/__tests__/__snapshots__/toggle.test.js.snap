// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app 1`] = `
<button
  onClick={[Function]}
>
  ON
</button>
`;

exports[`app 2`] = `
<button
  onClick={[Function]}
>
  OFF
</button>
`;
